src/transformers/generation/utils.py

```diff
@@ -3439,6 +3450,19 @@ def _sample(
         compile_forward = self._valid_auto_compile_criteria(model_kwargs, generation_config)
         if compile_forward:
             os.environ["TOKENIZERS_PARALLELISM"] = "0"
+            # If we use FA2 and a static cache, we cannot compile with fullgraph
+            if self.config._attn_implementation == "flash_attention_2" and isinstance(
+                model_kwargs.get("past_key_values"), (StaticCache, HybridCache, HybridChunkedCache)
```

Comment:
we can use the `is_compileable` cache attribute :D 

(it has a typo, it should be `is_compilable`, at some point we'll should do a deprecation cycle)